eng/pipelines/templates/steps/archetype-sdk-docs.yml

```diff
@@ -14,138 +14,19 @@ steps:
-            Copy-Item "$(Build.BinariesDirectory)/$($SafeName)/docfx-output/docfx_project/api/index.md" -Destination "$(Build.BinariesDirectory)/$($SafeName)/docfx-output/docfx_project/index.md"-Force
-            Write-Host "Package ReadMe was not found"
-          }
-
-          # Copy over generated yml and other assets
-          Write-Host "Copy over generated yml and other assets"
-          Copy-Item "$($SafeName)/dll-yaml-output/*"-Destination "$($SafeName)/docfx-output/docfx_project/api" -Recurse
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/assets/docfx.json" -Destination "$($SafeName)/docfx-output/docfx_project" -Recurse -Force
-          New-Item -Path "$($SafeName)/docfx-output/docfx_project" -Name templates -ItemType directory
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/templates/**" -Destination "$(Build.BinariesDirectory)/$($SafeName)/docfx-output/docfx_project/templates" -Recurse -Force
-
-          # Create Toc for Site Navigation
-          Write-Host "Create Toc for Site Navigation"
-          New-Item "$($SafeName)/docfx-output/docfx_project/toc.yml" -Force
-          Add-Content -Path "$($SafeName)/docfx-output/docfx_project/toc.yml" -Value "- name: ${{artifact.name}}`r`n  href: api/`r`n  homepage: api/index.md"
-
-          # Build Doc Content
-          Write-Host "Build Doc Content"
-          ./docfx/docfx.exe build "$($SafeName)/docfx-output/docfx_project/docfx.json"
-
-          # Copy over site Logo
-          Write-Host "Copy over site Logo"
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/assets/logo.svg" -Destination "$(Build.BinariesDirectory)/$($SafeName)/docfx-output/docfx_project/_site" -Recurse -Force
-
-        displayName: Build DocFx Documenttion Site for ${{artifact.name}}
-        workingDirectory: $(Build.BinariesDirectory)
-      - task: PublishPipelineArtifact@0
-        condition: succeeded()
-        inputs:
-          artifactName: Docs.${{artifact.name}}
-          targetPath: $(Build.BinariesDirectory)/$(artifactsafename)/docfx-output/docfx_project/_site
\ No newline at end of file
+  - ${{ each artifact in parameters.Artifacts }}:
+    - pwsh: |
+        . $(Build.SourcesDirectory)/doc/ApiDocGeneration/Generate-Api-Docs.ps1
+        Build-Docs -ArtifactName '${{artifact.name}}' -ServiceDirectory '${{parameters.ServiceDirectory}}' `
+        -ArtifactsSafeName '${{artifact.safeName}}' -ArtifactsDirectoryName '${{artifact.directoryName}}' `
+        -LibType '${{parameters.LibType}}' -RepoRoot $(Build.SourcesDirectory) -BinDirectory $(Build.BinariesDirectory)
+      displayName: Build DocFx Documenttion Site for ${{artifact.name}}
+    - task: PublishPipelineArtifact@0
+      condition: succeeded()
+      inputs:
+        artifactName: Docs.${{artifact.name}}
+        targetPath: $(Build.BinariesDirectory)/$(artifactsafename)/docfx-output/docfx_project/_site
```

Comment:
We should simply export the entire path from the PS script instead of making assumptions about the rest of this path here.

Reply:
What do you mean by `export the entire path`. Do you mean not to use the PublishPipelineArtifacts task? How do I do that?

Reply:
By export I mean set `echo "##vso[task.setvariable variable=artifactsafename]$($SafeName)"` However instead of just the safename the entire path instead.